xs/src/libslic3r/IO/TMF.cpp

```diff
@@ -0,0 +1,1066 @@
+#include "TMF.hpp"
+
+namespace Slic3r { namespace IO {
+
+TMFEditor::TMFEditor(std::string input_file, Model *model)
```

Comment:
I recommend an initialize list here (also review for anything that would be better to have as a constant), instead of a constructor like this. You can stick it in the hpp as well if it's just an initialize list.